RecoTracker/MkFitCore/interface/HitStructures.h

```diff
@@ -0,0 +1,874 @@
+
+  public:
+    const LayerInfo* m_layer_info = nullptr;
+    vecvecPhiBinInfo_t m_phi_bin_infos;
+    vecvecPhiBinDead_t m_phi_bin_deads;
+    std::vector<float> m_hit_phis;
+    std::vector<float> m_hit_qs;
+
+    float m_qmin, m_qmax, m_fq;
+    int m_nq = 0;
+    bool m_is_barrel;
+
+    int layer_id() const { return m_layer_info->m_layer_id; }
+    bool is_barrel() const { return m_is_barrel; }
+    bool is_endcap() const { return !m_is_barrel; }
+    int bin_index(int q, int p) const { return q * Config::m_nphi + p; }
+
+    PhiBinInfo_t operator[](int i) const {
+      int q = i / Config::m_nphi;
+      int p = i % Config::m_nphi;
+      return m_phi_bin_infos[q][p];
+    }
+
+    bool is_within_z_limits(float z) const { return m_layer_info->is_within_z_limits(z); }
+    bool is_within_r_limits(float r) const { return m_layer_info->is_within_r_limits(r); }
+
+    WSR_Result is_within_z_sensitive_region(float z, float dz) const {
+      return m_layer_info->is_within_z_sensitive_region(z, dz);
+    }
+
+    WSR_Result is_within_r_sensitive_region(float r, float dr) const {
+      return m_layer_info->is_within_r_sensitive_region(r, dr);
+    }
+
+    // Adding flag for mono/stereo
+    bool is_stereo_lyr() const { return m_layer_info->is_stereo_lyr(); }
+
+    // Adding info on sub-detector
+    bool is_pixb_lyr() const { return m_layer_info->is_pixb_lyr(); }
+    bool is_pixe_lyr() const { return m_layer_info->is_pixe_lyr(); }
+    bool is_pix_lyr() const { return m_layer_info->is_pix_lyr(); }
+    bool is_tib_lyr() const { return m_layer_info->is_tib_lyr(); }
+    bool is_tob_lyr() const { return m_layer_info->is_tob_lyr(); }
+    bool is_tid_lyr() const { return m_layer_info->is_tid_lyr(); }
+    bool is_tec_lyr() const { return m_layer_info->is_tec_lyr(); }
+
+    // Testing bin filling
+    static constexpr float m_fphi = Config::m_nphi / Config::TwoPI;
+    static constexpr int m_phi_mask = 0xff;
+    static constexpr int m_phi_bits = 8;
+    static constexpr float m_fphi_fine = 1024 / Config::TwoPI;
+    static constexpr int m_phi_mask_fine = 0x3ff;
+    static constexpr int m_phi_bits_fine = 10;  //can't be more than 16
+    static constexpr int m_phi_bits_shift = m_phi_bits_fine - m_phi_bits;
+    static constexpr int m_phi_fine_xmask = ~((1 << m_phi_bits_shift) - 1);
+
+  protected:
+#ifdef COPY_SORTED_HITS
+    void alloc_hits(int size) {
+      m_hits = (Hit*)std::aligned_alloc(64, sizeof(Hit) * size);
+      m_capacity = size;
+      for (int ihit = 0; ihit < m_capacity; ihit++) {
+        m_hits[ihit] = Hit();
+      }
+    }
+
+    void free_hits() { std::free(m_hits); }
+#endif
+
+    void setup_bins(float qmin, float qmax, float dq);
+
+    // Not used.
+    // void set_phi_bin(int q_bin, int phi_bin, uint16_t &hit_count, uint16_t &hits_in_bin)
+    // {
+    //   m_phi_bin_infos[q_bin][phi_bin] = { hit_count, hit_count + hits_in_bin };
+    //   hit_count  += hits_in_bin;
+    //   hits_in_bin = 0;
+    // }
```

Comment:
delete

Reply:
Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit: https://github.com/trackreco/cmssw/pull/76/commits/20e8958ee9ec6436b3be8a101b90d9f1b956f31c